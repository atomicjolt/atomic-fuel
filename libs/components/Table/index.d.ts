/// <reference types="react" />
import './styles.css';
export declare enum SortDirection {
    asc = "ASC",
    desc = "DESC"
}
export declare type ColumnType<T extends string | number> = {
    dataName: T;
    displayName: string;
    hidden?: boolean;
};
export interface Props<T extends string | number> {
    columns: ColumnType<T>[];
    defaultSortColumn: T;
    searchTerm: string;
    searchColumn: T;
    headClasses?: string;
    onSearch: (arg: string) => void;
    renderTableContent: (...arg: any[]) => any;
}
export declare const composeTable: <T extends string | number>() => ({ columns, searchTerm, searchColumn, headClasses, defaultSortColumn, onSearch, renderTableContent, }: Props<T>) => JSX.Element;
