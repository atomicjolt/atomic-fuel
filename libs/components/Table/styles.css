@keyframes spinner {
  0% {
    transform: rotate(0deg); }
  100% {
    transform: rotate(360deg); } }

.aj-table {
  width: 100%;
  border-collapse: collapse;
  text-align: left;
  table-layout: fixed;
  min-width: 800px;
  font-family: 'Lato', 'sans-serif'; }
  .aj-table__row {
    border-bottom: 1px solid #DDD;
    height: 45px; }
  .aj-table__head {
    font-size: 12px;
    font-weight: 700;
    cursor: pointer;
    border-bottom: 1px solid #DDD; }
    .aj-table__head-row {
      height: 44px; }
  .aj-table__flex-cell {
    display: flex;
    align-items: center; }
