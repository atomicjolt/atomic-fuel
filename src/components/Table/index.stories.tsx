import React from 'react';
import { Story, Meta } from '@storybook/react';
import StoryWrapper from '../StoryWrapper';
import { composeTable, Props } from '.';
import { TableRow } from '../TableRow';

enum FilterType {
	name = 'name',
	dueAt = 'dueAt',
	createdAt = 'createdAt',
	completed = 'completed',
}

const Table1 = composeTable<FilterType>();
export default {
  title: 'Table',
  component: Table1,
} as Meta;

const Template1: Story<Props<FilterType>> = (args) => (
  <StoryWrapper>
    <Table1 {...args} />
  </StoryWrapper>
);

export const Default = Template1.bind({});
const defaultColumns = [
	{ dataName: FilterType.name, displayName: 'NAME' },
	{ dataName: FilterType.dueAt, displayName: 'DUE' },
	{ dataName: FilterType.createdAt, displayName: 'CREATED' },
	{ dataName: FilterType.completed, displayName: 'COMPLETED' },
	{ dataName: 'actions', displayName: 'actions', hidden: true },
]

Default.args = {
	columns: defaultColumns,
	searchTerm: 'Hello World',
	searchColumn: FilterType.name,
	onSearch: () => console.log('Sup'),
	renderTableContent: (one = 'one', two = 2, three = 'III') => (
		<tbody>
			<tr>
				<td>{one}</td>
				<td>{two}</td>
				<td>{three}</td>
				<td>四</td>
				<td>E</td>
			</tr>
		</tbody>
	)
};

enum FilterType2 {
	product = 'product',
	calories = 'calories',
	fat = 'fat',
	sugar = 'sugar',
	protien = 'protien',
	carbs = 'carbs',
}

const Table2 = composeTable<FilterType2>();

const Template2: Story<Props<FilterType2>> = (args) => (
  <StoryWrapper>
    <Table2 {...args} />
  </StoryWrapper>
);

export const TableTwo = Template2.bind({});
const defaultColumns2 = [
	{ dataName: FilterType2.product, displayName: 'PRODUCT' },
	{ dataName: FilterType2.calories, displayName: 'CALORIES' },
	{ dataName: FilterType2.fat, displayName: 'FAT' },
	{ dataName: FilterType2.sugar, displayName: 'SUGAR' },
	{ dataName: FilterType2.protien, displayName: 'PROTIEN' },
	{ dataName: FilterType2.carbs, displayName: 'CARBS' },
]

TableTwo.args = {
	columns: defaultColumns2,
	searchTerm: 'Hello World',
	searchColumn: FilterType2.product,
	onSearch: () => console.log('Sup'),
	renderTableContent: (one = 'one', two = 2, three = 'III') => (
		<tbody>
			<tr>
				<td>{one}</td>
				<td>{two}</td>
				<td>{three}</td>
				<td>四</td>
				<td>E</td>
				<td>VI</td>
			</tr>
		</tbody>
	)
};