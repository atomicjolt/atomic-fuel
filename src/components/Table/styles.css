@keyframes spinner {
  0% {
    transform: rotate(0deg); }
  100% {
    transform: rotate(360deg); } }

.aj-table {
  width: 100%;
  border-collapse: collapse;
  text-align: left;
  table-layout: fixed;
  min-width: 800px; }
  .aj-table tr {
    border-bottom: 1px solid #DDD;
    height: 45px; }
  .aj-table thead .header-NAME {
    width: 50%;
    cursor: pointer; }
  .aj-table thead .header-DUE_AT {
    width: 10%;
    cursor: pointer; }
  .aj-table thead .header-CREATED_AT {
    width: 10%;
    cursor: pointer; }
  .aj-table thead .header-COMPLETED {
    width: 110px;
    cursor: pointer; }
