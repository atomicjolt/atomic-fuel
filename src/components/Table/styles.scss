@import "../../styles/styles.scss";

.aj-table {
	width: 100%;
	border-collapse: collapse;
	text-align: left;
	table-layout: fixed;
	min-width: 800px;

	tr {
		border-bottom: 1px solid $whiteDD;
		height: 45px;
	}

	thead .header {
		&-NAME {
			width: 50%;
			cursor: pointer;
		}

		&-DUE_AT {
			width: 10%;
			cursor: pointer;
		}

		&-CREATED_AT {
			width: 10%;
			cursor: pointer;
		}

		&-COMPLETED {
			width: 110px;
			cursor: pointer;
		}
	}
}