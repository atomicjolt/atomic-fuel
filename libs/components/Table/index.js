"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Table = exports.SortDirection = exports.TableFilter = void 0;

var _react = _interopRequireWildcard(require("react"));

var _TableHead = require("../TableHead");

require("./styles.css");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr == null ? null : typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]; if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

var TableFilter;
exports.TableFilter = TableFilter;

(function (TableFilter) {
  TableFilter["createdAt"] = "CREATED_AT";
  TableFilter["dueAt"] = "DUE_AT";
  TableFilter["name"] = "NAME";
  TableFilter["completed"] = "COMPLETED";
})(TableFilter || (exports.TableFilter = TableFilter = {}));

var SortDirection;
exports.SortDirection = SortDirection;

(function (SortDirection) {
  SortDirection["asc"] = "ASC";
  SortDirection["desc"] = "DESC";
})(SortDirection || (exports.SortDirection = SortDirection = {}));

var Table = function Table(_ref) {
  var columns = _ref.columns,
      searchTerm = _ref.searchTerm,
      searchColumn = _ref.searchColumn,
      headClasses = _ref.headClasses,
      onSearch = _ref.onSearch,
      renderTableContent = _ref.renderTableContent;

  var _useState = (0, _react.useState)(TableFilter.createdAt),
      _useState2 = _slicedToArray(_useState, 2),
      sortColumn = _useState2[0],
      setSortColumn = _useState2[1];

  var _useState3 = (0, _react.useState)(SortDirection.asc),
      _useState4 = _slicedToArray(_useState3, 2),
      sortDirection = _useState4[0],
      setSortDirection = _useState4[1];

  return /*#__PURE__*/_react["default"].createElement("table", {
    className: "aj-table"
  }, /*#__PURE__*/_react["default"].createElement(_TableHead.TableHead, {
    columns: columns,
    classes: headClasses,
    searchColumn: searchColumn,
    sortColumn: sortColumn,
    sortDirection: sortDirection,
    searchTerm: searchTerm,
    onSearch: onSearch,
    onSort: function onSort(newSortDirection, newSortColumn) {
      setSortDirection(newSortDirection);
      setSortColumn(newSortColumn);
    }
  }), renderTableContent());
};

exports.Table = Table;