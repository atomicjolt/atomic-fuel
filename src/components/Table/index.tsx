import React, { useState } from 'react';
import { composeTableHead } from '../TableHead';
import './styles.css';

export enum SortDirection {
  asc = 'ASC',
  desc = 'DESC',
}

export type ColumnType<T extends string | number> = {
	dataName: T
	displayName: string
	hidden?: boolean
}

export interface Props<T extends string | number> {
	columns: ColumnType<T>[]
	defaultSortColumn: T
	searchTerm: string
	searchColumn: T
	headClasses?: string
	onSearch: (arg: string) => void
	renderTableContent: (...arg: any[]) => any
}

export const composeTable = <T extends string | number>() => {
	const TableHead = composeTableHead<T>();
	return ({
		columns,
		searchTerm,
		searchColumn,
		headClasses,
		defaultSortColumn,
		onSearch,
		renderTableContent,
	}: Props<T>) => {
		const [sortColumn, setSortColumn] = useState<T>(defaultSortColumn);
		const [sortDirection, setSortDirection] = useState(SortDirection.asc);

		return (
			<table className="aj-table">
				<TableHead
					columns={columns}
					classes={headClasses}
					searchColumn={searchColumn}
					sortColumn={sortColumn}
					sortDirection={sortDirection}
					searchTerm={searchTerm}
					onSearch={onSearch}
					onSort={(newSortDirection: SortDirection, newSortColumn: T) => {
						setSortDirection(newSortDirection);
						setSortColumn(newSortColumn);
					}}
				/>
				{renderTableContent()}
			</table>
		);
	}
}