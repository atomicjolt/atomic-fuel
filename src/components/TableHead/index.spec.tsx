import React from 'react';
import { render } from '@testing-library/react';
import { SortDirection } from '../Table';
import { TableHead } from '.';

jest.mock('../SortableHeader/utils', () => ({
  getID: () => 123
}))

describe('TableHead', () => {
  it('Should match default Snapshot', () => {
    const { asFragment } = render(
      <table>
        <TableHead
          columns={[{ dataName: 'name', displayName: 'Name' }]}
          onSearch={() => {}}
          onSort={() => {}}
          searchColumn="name"
          searchTerm=""
          sortColumn="name"
          sortDirection={SortDirection.asc}
        />
      </table>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
