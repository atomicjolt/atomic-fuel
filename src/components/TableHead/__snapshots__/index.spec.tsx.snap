// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHead Should match default Snapshot 1`] = `
<DocumentFragment>
  <table>
    <thead
      class="aj-table__head"
    >
      <tr
        class="aj-table__head-row"
      >
        <th
          aria-label="ascending"
          class="aj-table__head-row"
          scope="col"
        >
          <div
            class="aj-table__flex-cell"
          >
            <span>
              Name
            </span>
            <button
              class="aj-btn aj-btn--sort"
              type="button"
            >
              <i
                aria-hidden="true"
                class="material-icons-outlined sort-direction"
              >
                arrow_drop_up
              </i>
              <i
                aria-hidden="true"
                class="material-icons-outlined "
              >
                arrow_drop_down
              </i>
            </button>
            <div
              class="aj-input aj-input--search"
              id="container-123"
            >
              <label
                class="aj-hidden"
                for="search-123"
              />
              <input
                aria-controls="status-123"
                id="search-123"
                tabindex="-1"
                value=""
              />
              <div
                aria-live="polite"
                class="aj-hidden"
                id="status-123"
                role="status"
              />
              <button
                aria-controls="container-123"
                aria-expanded="false"
                type="button"
              >
                <i
                  aria-hidden="true"
                  class="material-icons-outlined"
                >
                  search
                </i>
              </button>
            </div>
          </div>
        </th>
      </tr>
    </thead>
  </table>
</DocumentFragment>
`;
